src/mail/editor/MailEditor.ts

```diff
@@ -139,14 +152,30 @@ export class MailEditor implements Component<MailEditorAttrs> {
 		const model = a.model
 		this.sendMailModel = model
 		this.templateModel = a.templateModel
+		this.blockExternalContent = a.doBlockExternalContent()
 
 		// if we have any CC/BCC recipients, we should show these so, should the user send the mail, they know where it will be going to
 		this.areDetailsExpanded = model.bccRecipients().length + model.ccRecipients().length > 0
 
 		this.editor = new Editor(200, (html, isPaste) => {
 			const sanitized = htmlSanitizer.sanitizeFragment(html, {
-				blockExternalContent: !isPaste && a.doBlockExternalContent(),
+				blockExternalContent: !isPaste && this.blockExternalContent,
 			})
+
+			if (a.doBlockExternalContent()) {
+				model.getExternalImageRule().then(async (externalImageRule) => {
```

Comment:
why is this lambda marked as `async`?

Reply:
we expect this to be executed before we invoke the sanitizer, right?

Reply:
The async was needed before but can be taken away now

And the function depends on the externalContent count, if there's no external content we don't need to block anything

Reply:
I still didn't get it sorry. We are here inside the sanitizing function invocation by the editor. Before we return we start to fetch the image rules async to set `this.blockExternalContent`. Do we do this to show the banner? If yes, should we call `m.redraw()`?

Reply:
Also, why do we do this here, inside the sanitizing function?

Reply:
We need to check the rules to show the image, but yes, it should be outside the sanitize function. We can call it inside editor initialization to prevent the editor to register a change and save a lot of drafts just for applying the rule